### PR TITLE
Update method names for client registrations

### LINKED ISSUE #1510: Consider renaming component or resource methods to disambiguate and reduce confusion
I am following the steps from:
[https://learn.microsoft.com/en-us/dotnet/aspire/messaging/azure-service-bus-component?tabs=dotnet-cli](url)

However I am not able to add ServiceBus (screen below). 
image

Text:
`Severity	Code	Description	Project	File	Line	Suppression State	Details
Error	CS1061	'IDistributedApplicationBuilder' does not contain a definition for 'AddAzureServiceBus' and no accessible extension method 'AddAzureServiceBus' accepting a first argument of type 'IDistributedApplicationBuilder' could be found (are you missing a using directive or an assembly reference?)	AspireStarter2.AppHost	G:\Aspire\AspireStarter2\AspireStarter2.AppHost\Program.cs	4	Active	
`

Thanks for input

### PR DESCRIPTION
**Pull request by sebastienros**
Fixes #1510

We may consider different names based on taste. Sometimes the name matches the client type, and sometimes not.

Examples:

- `AddRabbitMQClient()`, maybe remove the `MQ`. The client name is `IConnection`.
- `AddRedisClient()`, the client name is `IConnectionMultiplexer`, I think it's ok.
- `AddCosmosClient()`, the client name is `CosmosClient` but might sound better with `AddCosmosDBClient`.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/2630)

### CODE REVIEW

File: playground/AzureSearchEndToEnd/AzureSearch.ApiService/Program.cs [hunk 1]
```diff
@@ -12,7 +12,7 @@
 var builder = WebApplication.CreateBuilder(args);
 
 builder.AddServiceDefaults();
-builder.AddAzureSearch("search");
+builder.AddSearchIndexClient("search");
```
Comment (DamianEdwards) [inline playground/AzureSearchEndToEnd/AzureSearch.ApiService/Program.cs:15]: I think we'd want `Azure` on all these still right?

**Comment by davidfowl** [comment]
Do we want to add new ones and deprecate the old ones?

File: playground/AzureSearchEndToEnd/AzureSearch.ApiService/Program.cs (hunk 1 above)
Comment (sebastienros) [reply playground/AzureSearchEndToEnd/AzureSearch.ApiService/Program.cs:15]: Maybe, https://github.com/dotnet/aspire/issues/1510#issuecomment-1965571299

Do we need a meeting or wait for more opinions here?

**Comment by sebastienros** [comment]
> Do we want to add new ones and deprecate the old ones?

So ship a preview (at least) with deprecations, to provide a hint about how to update existing apps? I like it.

File: playground/AzureSearchEndToEnd/AzureSearch.ApiService/Program.cs (hunk 1 above)
Comment (DamianEdwards) [reply playground/AzureSearchEndToEnd/AzureSearch.ApiService/Program.cs:15]: Let's get some opinions here.

**Comment by davidfowl** [comment]
Yep!

**Comment by davidfowl** [comment]
> So ship a preview (at least) with deprecations, to provide a hint about how to update existing apps? I like it.

Yep.

**Comment by eerhardt** [comment]
> So ship a preview (at least) with deprecations, to provide a hint about how to update existing apps?

Make sure we log an issue in the correct milestone to remove these deprecated methods. We don't want to ship GA with them.

**Comment by davidfowl** [comment]
https://github.com/dotnet/aspire/issues/2417

File: playground/AzureSearchEndToEnd/AzureSearch.ApiService/Program.cs (hunk 1 above)
Comment (davidfowl) [reply playground/AzureSearchEndToEnd/AzureSearch.ApiService/Program.cs:15]: Agree Azure in the names

**Comment by sebastienros** [comment]
I pushed a brand new set of changes with the feedback I got. Now it keeps the same prefixes as before with the added "Client" suffix when necessary. And the existing one is marked as [Obsolete].

File: playground/bicep/BicepSample.ApiService/Program.cs [hunk 2]
```diff
@@ -16,12 +16,12 @@
...
-builder.AddRedis("redis");
-builder.AddAzureBlobService("blob");
-builder.AddAzureTableService("table");
-builder.AddAzureQueueService("queue");
-builder.AddAzureServiceBus("sb");
+builder.AddAzureCosmosDbClient("cosmos");
+builder.AddRedisClient("redis");
+builder.AddAzureBlobServiceClient("blob");
+builder.AddAzureTableServiceClient("table");
+builder.AddAzureQueueServiceClient("queue");
```
Comment (DamianEdwards) [inline playground/bicep/BicepSample.ApiService/Program.cs:23]: What do folks think about shortening these?

```suggestion
builder.AddAzureBlobsClient("blob");
builder.AddAzureTablesClient("table");
builder.AddAzureQueuesClient("queue");
```

Comment (eerhardt) [reply playground/bicep/BicepSample.ApiService/Program.cs:23]: The underlying class names are:

* `Azure.Storage.Blobs.BlobServiceClient`
* `Azure.Data.Tables.TableServiceClient`
* `Azure.Storage.Queues.QueueServiceClient`

Is it confusing if we drop "Service" from these names? In general I like to stick with the naming of the underlying thing. But agree that @DamianEdwards's suggestion looks cleaner.

Comment (davidfowl) [reply playground/bicep/BicepSample.ApiService/Program.cs:23]: Keep the underlying names, that way we don't need to think too much 😄

Comment (DamianEdwards) [reply playground/bicep/BicepSample.ApiService/Program.cs:23]: FWIW we're adding `Client` to all the methods which happens to match the underlying types in this case but not all cases.

Comment (davidfowl) [reply playground/bicep/BicepSample.ApiService/Program.cs:23]: I won’t die on this hill but we should optimize for less case by case naming conventions.

I could go with either.

Comment (DamianEdwards) [reply playground/bicep/BicepSample.ApiService/Program.cs:23]: I think we should aim to go with something generic like `Add[LogicalName]Client`, e.g. `AddRedisClient`, `AddAzureBlobsClient`, `AddSqlClient`, etc. (obvious exception for EF Core related methods that operate on a `DbContext`).

Comment (davidfowl) [reply playground/bicep/BicepSample.ApiService/Program.cs:23]: Lets close then.

Comment (sebastienros) [reply playground/bicep/BicepSample.ApiService/Program.cs:23]: > Lets close then.

Is that "Just do it"?

Comment (sebastienros) [reply playground/bicep/BicepSample.ApiService/Program.cs]: Done, also changed AddAzureKeyVault~Secrets~Client

**Review by davidfowl — APPROVED** [review]

**Pull request merged**